### PR TITLE
Allows for play again, fix joining game late bug, and shows players role at the end

### PR DESCRIPTION
**Pull request by shandyk18**
When a game ends, players are redirected to the waiting room and shown results.

Screenshot 2020-08-14 at 9 25 06 AM - Display 2

### CODE REVIEW

File: ubg/src/common/WaitingRoom.js [hunk 1]
```diff
@@ -75,39 +77,65 @@ function WaitingRoom({gameName, gameDescription, leaveRoom,
...
         </Typography>
       </div>
+      <div>
+        {
+          win && win !== 0 ?
+          (
+            <Modal
+              open={endOpen}
+              onClose={() => setEndOpen(false)}
+              aria-labelledby="simple-modal-title"
```
Comment (danieltre23) [inline ubg/src/common/WaitingRoom.js:87]: Do we need this?

File: ubg/src/common/WaitingRoom.js [hunk 2]
```diff
@@ -75,39 +77,65 @@ function WaitingRoom({gameName, gameDescription, leaveRoom,
...
+                        onClick={() => {
+                          gameId === '925' && setSettingsOpen(true);
+                        }}
+                      >
+                        Start Game
+                      </Button>
+                      <Modal
+                        open={settingsOpen}
+                        onClose={() => setSettingsOpen(false)}
+                        aria-labelledby="simple-modal-title"
```
Comment (danieltre23) [inline ubg/src/common/WaitingRoom.js:122]: Same here

File: ubg/src/pages/Room.js [hunk 3]
```diff
@@ -267,6 +282,20 @@ function Room({setUsersData, setCurrentUser, setRoomData}) {
...
   }
 
+  /**
+   * Sets up player to be able to play again
+   */
+  function playAgain() {
+    usersCollection.doc(user.uid).update({
+      alive: true,
+      chose: false,
+      order: Math.floor(Math.random() * 20),
```
Comment (danieltre23) [inline ubg/src/pages/Room.js:292]: What are we using order for?

File: ubg/src/pages/Room.js [hunk 4]
```diff
@@ -267,6 +282,20 @@ function Room({setUsersData, setCurrentUser, setRoomData}) {
...
+    });
+    room.update({
+      started: false,
```
Comment (danieltre23) [inline ubg/src/pages/Room.js:295]: When the host starts a game doesn't that set all users to alive, chose: false?

Comment (shandyk18) [reply ubg/src/pages/Room.js:295]: Only in joinRoom() does anything for individual players themselves get set. startGame() only updates for the fields in the room only. That's why it is necessary to update this here.

File: ubg/src/pages/Room.js [hunk 5]
```diff
@@ -289,129 +318,149 @@ function Room({setUsersData, setCurrentUser, setRoomData}) {
...
+                          >
+                            <ChatIcon className={classes.sideMargin10px}/>
+                            <Typography variant="h6">Mafia</Typography>
+                          </div> :
+                          null
+                        }
+                      </div>
+                      <Chat
+                        disabled={(mafiaChatSelected && roomData.day) ||
+                          !inGame}
```
Comment (danieltre23) [inline ubg/src/pages/Room.js:416]: Isn't it the same to say '&& inGame'?

File: ubg/src/pages/Room.js (hunk 3 above)
Comment (shandyk18) [reply ubg/src/pages/Room.js:292]: Per game, everyone is given a random number (order). When the game starts, we assign roles based on what number they received.

File: ubg/src/pages/Room.js (hunk 4 above)
Comment (danieltre23) [reply ubg/src/pages/Room.js:295]: Okay cool

File: ubg/src/pages/Room.js (hunk 3 above)
Comment (danieltre23) [reply ubg/src/pages/Room.js:292]: Cool cool cool

**Pull request merged**